falcon/request.py

```diff
@@ -2334,6 +2334,90 @@ def get_param_as_json(
 
         return val
 
+    def get_param_as_dict(
+        self,
+        name: str,
+        required: bool = False,
+        deep_object: bool = False,
+        store: StoreArg = None,
+        default: Any | None = None,
+    ) -> Any:
```

Comment:
this is likely `dict[str, str | None] | None` even if we don't want to do the overload like the other ones